pygitguardian/models.py

```diff
@@ -485,6 +485,69 @@ def __repr__(self) -> str:
         return f"content:{repr(self.content)}"
 
 
+class HoneytokenResponseSchema(BaseSchema):
+    id = fields.Int()
+    name = fields.String()
+    description = fields.String(allow_none=True)
+    created_at = fields.AwareDateTime()
+    status = fields.String()
+    triggered_at = fields.AwareDateTime(allow_none=True)
+    revoked_at = fields.AwareDateTime(allow_none=True)
+    open_events_count = fields.Int(allow_none=True)
+    type_ = fields.String(data_key="type")
+    creator_id = fields.Int(allow_none=True)
+    revoker_id = fields.Int(allow_none=True)
+    creator_api_token_id = fields.String(allow_none=True)
+    revoker_api_token_id = fields.String(allow_none=True)
+    token = fields.Mapping(key=fields.String(), value=fields.String())
+    tags = fields.List(fields.String())
+
+    @post_load
+    def make_honeytoken_response(
+        self, data: Dict[str, Any], **kwargs: Any
+    ) -> "HoneytokenResponse":
+        return HoneytokenResponse(**data)
+
+
+class HoneytokenResponse(Base):
+    """
+    honeytoken creation in the GitGuardian API.
+    Allow you to create and get a honeytoken.
+    Example:
+        {
+            "id": 141,
+            "name": "honeytoken A",
+            "description": "honeytoken used in the repository AA",
+            "created_at": "2019-08-22T14:15:22Z",
+            "status": "active",
+            "triggered_at": "2019-08-22T14:15:22Z",
+            "revoked_at": "2019-08-22T14:15:22Z",
+            "open_events_count": 122,
+            "type": "AWS",
+            "creator_id": 122,
+            "revoker_id": 122,
+            "creator_api_token_id": null,
+            "revoker_api_token_id": null,
+            "token": {
+                "access_token_id": "AAAA",
+                "secret_key": "BBB"
+            },
+        "tags": ["publicly_exposed"]
+        }
+    """
+
+    SCHEMA = HoneytokenResponseSchema()
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__()
+        for key, value in kwargs.items():
+            if not hasattr(self, key):
+                setattr(self, key, value)
```

Comment:
I am not fond of using kwargs this way because users or tools like mypy cannot know which arguments are valid. I know it's tedious to type, but I'd prefer it if the method explicitly listed its arguments.

Alternatively, you can use the approach from `iac_models.py` which uses `marshmallow_dataclass` to reduce much of the boilerplate.